pkg/util/deployment/deployment.go

```diff
@@ -40,6 +45,29 @@ const (
 	RollbackDone              = "DeploymentRollback"
 )
 
+// IsScalingEvent checks whether the provided deployment has been updated with a scaling event.
+func IsScalingEvent(d extensions.Deployment) bool {
+	surgeAnnotation, ok := d.Annotations[SurgePodsAnnotation]
+	if !ok {
+		// We are not sure so we should fallback to the rollout logic
+		return false
+	}
+	surge, err := strconv.Atoi(surgeAnnotation)
+	if err != nil {
+		// Messed-up annotation; fallback to the rollout logic
+		return false
+	}
+	maybeMaxSurge := 0
+	if d.Spec.Strategy.Type == extensions.RollingUpdateDeploymentStrategyType {
```

Comment:
This doesn't quite work, as noted above. Also, we don't need to use an annotation on Deployment. We could just use a field.